### ISSUE TITLE
Add automation for all tektoncd repos

### ISSUE THREAD

**Issue by bobcatfish** [tektoncd/plumbing#3]
# Expected Behavior

All of the repos in tektoncd should have a minimum of prow automation applied.

We should have tide automation for:
* [x] plumbing
* [x] dashboard
* [x] cli
* [x] experimental (OWNERS files in each subdir)
* [x] website
* [x] catalog

We should add unit test support for:
* [x] dashboard (https://github.com/tektoncd/dashboard/pull/72)
* [x] experimental (in each subdir?) : https://github.com/tektoncd/plumbing/pull/37 and https://github.com/tektoncd/experimental/pull/60
* [x] catalog
* [x] cli (https://github.com/tektoncd/cli/pull/14) – a bit more than unit tests

Unit test support is going to be interesting, the easiest way to add this initially is probably to copy-paste the prow jobs as is and add exactly the script the knative-tests image expects:

https://github.com/tektoncd/plumbing/blob/2ff52652ad654984139fd24666d7d557ddda8e75/prow/config.yaml#L107-L108

OR we could be adventurous and try using our own script.

OR be _very_ adventurous and try our own image.

# Actual Behavior

Pipelines has all the bells and whistles, the rest of the projects seem to have only a random smattering (e.g. community has tide merging, but this repo doesn't).

# Additional Info

* Each project may decide to add additional automation (e.g. integration tests, coverage) which is totally cool, hopefully setting up a minimum should give them a starting point.
* The work to do here is basically copy paste [in the prow config](https://github.com/tektoncd/plumbing/blob/master/prow/config.yaml) to copy the `unit-test` section from pipelines and duplicate it for the other repos.

**Comment by bobcatfish** [comment]
I also had to add branch protection rules to all the repos to make sure the tide check was required. Will need to back and add a requirement about the `tide` clause to repos that didn't have any open PRs:

- experimental
- website
- cli

**Comment by vdemeester** [comment]
@bobcatfish updated the requirements for `tide` (`dashboard`, `cli` and `experimental`) and builds checks for `dashboard` and `cli`. (`website` needs at least a PR before we can see the check :sweat_smile: )

For `experimental`, I was wondering how should be do that :thinking: each time a folder is created, we create a configuration here ? :thinking:

**Comment by vdemeester** [comment]
Also, the merging options are all the same (as `tektoncd/pipeline`).

![2019-04-24-175300](https://user-images.githubusercontent.com/6508/56674223-eba5b800-66b9-11e9-92b6-2038b9b86d7b.png)

**Comment by bobcatfish** [comment]
> For experimental, I was wondering how should be do that 🤔 each time a folder is created, we create a configuration here ? 🤔

A couple ideas:
- Have a top level script that looks for scripts in all subdirs
- Have a different prowjob for each folder (probably a bad idea)

Another issue is that when code in a dir changes, we probably want to _only_ run the tests for that dir - but we can always deal with this iteratively XD

**Comment by vdemeester** [comment]
Once https://github.com/tektoncd/plumbing/pull/37 is in, I think we can close this one :angel: :tada: 
Automation of `tektoncd/plumbing` itself is tracked by https://github.com/tektoncd/plumbing/issues/1

**Comment by vdemeester** [comment]
#37 is merged :dancer: 
/close

**Comment by bobcatfish** [comment]
Thanks @vdemeester !!!! :heart: :heart: :heart:
